### DISCUSSION TITLE
linear extrude behavior differs from openscad

### DISCUSSION THREAD

**Discussion by pca006132** [elalish/manifold#632] [General]
OpenSCAD performs rotate before scaling, while we do scaling before rotate, so some examples that uses both twist and scale behaves differently in openscad and manifoldcad. Also, the twist direction seems to be different.

Should we keep this behavior or change to match the behavior in openscad? Not sure if anyone uses both with manifold for now.

**Comment by hrgdavor** [comment]
I would vote to align in this relatively early stage.

**Comment by pca006132** [comment]
and they also have this flip diagonal function that chooses the diagonal with the shorter length, so it behaves better when the twist angle is large comparing with the number of slices.

**Reply by elalish** [reply]
Agreed, may as well align now. And yes, cutting the diagonal the right way is always a good idea.
